solutions/src/main/java/ru/sberbank/school/task11/Streams.java

```diff
@@ -0,0 +1,172 @@
+                    result.add(it);
+                }
+            }
+            return list = result;
+        };
+        operations.add(operation);
+        return this;
+    }
+
+    /**
+     * Преобразование элементов в какие-то другие элементы. Необходимо подобрать нужый интерфейс для передачи в этот метод.
+     * <p>
+     * Например, данный метод должен уметь извлекать какие-то поля из объектов исходного типа (если это возможно).
+     * <p>
+     * Аналог Stream#map.
+     * <p>
+     * Intermediate операция.
+     *
+     * @param function - правило траснформации элементов.
+     * @return стрим
+     */
+    public <R> Streams<T> transform(@NonNull Function<T, R> function) {
+        Supplier<List> operation = () -> {
+            List<R> result = new ArrayList<>();
+            for (T it : list) {
+                result.add(function.apply(it));
+            }
+            return result;
+        };
+        operations.add(operation);
+        return this;
+    }
+
+    /**
+     * Сортировка элементов. Необходимо подобрать нужый интерфейс для передачи в этот метод.
+     * <p>
+     * Данный метод должен уметь сортировать элементы по заданному правилу.
+     * <p>
+     * Аналог Stream#sorted
+     * <p>
+     * Intermediate операция.
+     *
+     * @param comparator - правило сортировки элементов.
+     * @return стрим
+     */
+    public Streams<T> sorted(@NonNull Comparator<? super T> comparator) {
+        Supplier<List> operation = () -> {
+            List<T> sorted = new ArrayList<>(list);
+            sorted.sort(comparator);
+            return sorted;
+        };
+        operations.add(operation);
+        return this;
+    }
+
+    /**
+     * Преобразование стрима в Map. Необходимо подобрать нужные интерфейсы для передачи в этот метод.
+     * <p>
+     * Данный метод должен уметь преобразовать стрим в Map используя keyMapper и valueMapper.
+     * <p>
+     * Terminate операция.
+     *
+     * @param keyMapper   - правило создания ключа.
+     * @param valueMapper - правило создания значения.
+     * @return Map, собранная по правилам keyMapper и valueMapper
+     */
+    public <K, V> Map<K, V> toMap(@NonNull Function<T , K> keyMapper,@NonNull  Function<T , V> valueMapper) {
+        List<T> resultList = doOperations();
+        Map<K, V> map = new HashMap<>();
+        for (T it : resultList) {
+            map.put(keyMapper.apply(it), valueMapper.apply(it));
+        }
+        return map;
+    }
+
+    /**
+     * Преобразование стрима в Set.
+     * <p>
+     * Данный метод должен уметь преобразовать стрим в Set. Обратите внимание, что метод sorted должен учитываться
+     * при вызове этого метода.
+     * <p>
+     * Terminate операция.
+     *
+     * @return Set элементов
+     */
+    public Set toSet() {
+        return new LinkedHashSet(doOperations());
+    }
+
+    /**
+     * Преобразование стрима в List.
+     * <p>
+     * Данный метод должен уметь преобразовать стрим в List.
+     * <p>
```

Comment:
&lt;p&gt; tag should be preceded with an empty line.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-151678185>*